File: storagemanager/src/main/java/com/example/alexeyglushkov/cachemanager/StorageMetadata.java
package com.example.alexeyglushkov.cachemanager;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * Created by alexeyglushkov on 02.05.16.
 */
public interface StorageMetadata {
    void setCreateTime(long createTime);
    long getCreateTime();

    void setExpireTime(long expireTime);
    long getExpireTime();
    boolean isExpired();

    void setContentSize(long size);
    long getContentSize();

    void setEntryClass(@NonNull Class cl);
    @Nullable Class getEntryClass();
}

File: storagemanager/src/main/java/com/example/alexeyglushkov/cachemanager/StorageMetadata.kt
package com.example.alexeyglushkov.cachemanager

/**
 * Created by alexeyglushkov on 02.05.16.
 */
interface StorageMetadata {
    var createTime: Long
    var expireTime: Long
    var contentSize: Long
    var entryClass: Class<*>?

    fun isExpired(): Boolean
}